### ISSUE TITLE
`ActorSystem.Create` does not work with .NET 5 single-file publish. 

### ISSUE THREAD

**Issue by andersstorhaug** [akkadotnet/akka.net#4876]
With Akka v1.4.18, .NET 5 and `PublishSingleFile` enabled, I get the following exception:

```
System.Configuration.ConfigurationErrorsException: Configuration system failed to initialize
 ---> System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
   at System.Reflection.RuntimeModule.GetFullyQualifiedName()
   at System.Reflection.RuntimeModule.get_Name()
   at System.Configuration.ClientConfigPaths..ctor(String exePath, Boolean includeUserConfig)
   at System.Configuration.ClientConfigPaths.GetPaths(String exePath, Boolean includeUserConfig)
   at System.Configuration.ClientConfigurationHost.get_ConfigPaths()
   at System.Configuration.ClientConfigurationHost.GetStreamName(String configPath)
   at System.Configuration.ClientConfigurationHost.get_IsAppConfigHttp()
   at System.Configuration.Internal.DelegatingConfigHost.get_IsAppConfigHttp()
   at System.Configuration.ClientConfigurationSystem..ctor()
   at System.Configuration.ConfigurationManager.EnsureConfigurationSystem()
   --- End of inner exception stack trace ---
   at System.Configuration.ConfigurationManager.EnsureConfigurationSystem()
   at System.Configuration.ConfigurationManager.PrepareConfigSystem()
   at System.Configuration.ConfigurationManager.GetSection(String sectionName)
   at Akka.Configuration.ConfigurationFactory.Load()
   at Akka.Actor.ActorSystem.Create(String name, ActorSystemSetup setup)
```

See [`ActorSystem.cs` L262](https://github.com/akkadotnet/akka.net/blob/1.4.18/src/core/Akka/Actor/ActorSystem.cs#L262):

```csharp
var appConfig = bootstrapSetup.FlatSelect(_ => _.Config).GetOrElse(ConfigurationFactory.Load());
```

I believe that this issue could be resolved by making the `ConfigurationFactory.Load()` call here lazy, as right now it will execute no matter what. I see that the underlying implementation is guarded by `#if CONFIGURATION`, however this is problematic when compiling with a Windows RID and single file.

The underlying issue AFAIK is that `System.Configuration` is effectively not supported with .NET 5 single file publish.

A short-term workaround is to build with `/p:IncludeAllContentForSelfExtract=true`.

**Comment by Arkatufus** [comment]
@andersstorhaug 
I can't reproduce this using the last 1.4.18 release. Can you test that this simple example also work on your system?
https://github.com/Arkatufus/Akka.Net5.Single.File.Publish.Example

**Comment by Arkatufus** [comment]
You might want to compare the .pubxml file to see if there are any discrepancies

**Comment by andersstorhaug** [comment]
I'm thinking that this issue may be specific to the usage of `BootstrapSetup`.

I'll use your example and see if I can reproduce that way.

**Comment by andersstorhaug** [comment]
Yep, that's it. Relevant change (both projects):

```csharp
var setup = BootstrapSetup.Create().WithConfig(config);

using (var system = ActorSystem.Create("MyServer", setup))
```

**Comment by Arkatufus** [comment]
Ah, ok, I'll give it another look see, thanks :)

**Comment by Arkatufus** [comment]
@andersstorhaug 
Well, I still can't reproduce it on my system. A few question for you: 
- Do you have self contained, publish ready to run, or publish trimmed turned on?
- I'm assuming that the .NET 5.0 runtime is installed?
- Can you build the project manually from command prompt/powershell using this command: `dotnet publish -p:PublishProfile=./Properties/PublishProfiles/FolderProfile.pubxml -p:PublishReadyToRunShowWarnings=true` and when you do, do you see any warnings related to `System.Configuration.ConfigurationManager` in the log?
- Can you try and install the latest `System.Configuration.ConfigurationManager` nuget package and try again?

**Comment by Arkatufus** [comment]
Another thing that might help is what I personally call "nuking the obj and bin folders", delete the bin and obj folder from your project folder, re-run "dotnet restore", and build the project again. Sometimes all you need to do is to start fresh.

**Comment by andersstorhaug** [comment]
Strange.. no warnings with `-p:PublishReadyToRunShowWarnings=true`. Reproduced this on my other machine as well. And, if I disable single-file publish, the executables run fine.

I opened a PR to your example repo just to make sure we're running the same code.

**Comment by andersstorhaug** [comment]
@Arkatufus Just out of curiosity can you reproduce the issue with this repository? Details in `README.md`.

https://github.com/andersstorhaug/Net5SingleFileConfigurationManager

**Comment by andersstorhaug** [comment]
Here's the underlying issue:

https://github.com/dotnet/runtime/issues/42161

Won't be fixed in .NET 5.

https://github.com/dotnet/runtime/pull/42306#issuecomment-703933121

**Comment by andersstorhaug** [comment]
It looks like this was fixed in SDK 5.0.202... which was released on Tuesday.

What timing :).
